### DISCUSSION TITLE
How to NOT display BubbleMenu for certain nodes

### DISCUSSION THREAD

**Discussion by cadars** [ueberdosis/tiptap#2059] [General Questions] [answered]
I am aware of `shouldShow`, but I guess what I am after would be something like `shouldNotShow`?

(My use case is `pre` code blocks (where no other formatting is possible) and selected horizontal rules.)

**Comment by philippkuehn** [comment, marked as answer]
You can exactly do that with `shouldShow`.

**Reply by cadars** [reply]
Thanks! So I should list ALL the nodes and marks with `shouldShow`, except the few I don't want? That's what I wanted to avoid, but OK :)

**Reply by philippkuehn** [reply]
It’s a callback method where you can put custom logic. You don’t have to list all nodes.

**Reply by cadars** [reply]
My programming skills are quite limited, but I’ll look into that, thank you!

**Reply by philippkuehn** [reply]
look at the default `shouldShow` method for a starting point
https://github.com/ueberdosis/tiptap/blob/33420f4ae06d0d7eec16201a2e650be83cbb9de9/packages/extension-bubble-menu/src/bubble-menu-plugin.ts#L43-L58
